gapis/api/transform2/command_id.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package transform2 contains the elements to be able to transform
+// commands which consist of interfaces for individual transform operations
+// and a transform chain to run all of them.
+package transform2
+
+import (
+	"github.com/google/gapid/gapis/api"
+)
+
+type CommandType uint32
+
+const (
+	BeginCommand     CommandType = 0
+	TransformCommand CommandType = 1
+	EndCommand       CommandType = 2
+)
+
+type CommandID struct {
+	id          api.CmdID
+	commandType CommandType
+}
+
+func NewCommandID(id api.CmdID, cmdType CommandType) CommandID {
+	return CommandID{
+		id:          id,
+		commandType: cmdType,
+	}
+}
+
+func (c *CommandID) GetID() api.CmdID {
+	return c.id
```

Comment:
Should panic, error or do something else if it's BeginCommand or EndCommand

Reply:
Same with the last comment. But I will take a look at this.